omero_marshal/encode/encoders/well.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Glencoe Software, Inc. All rights reserved.
+#
+# This software is distributed under the terms described by the LICENCE file
+# you can find at the root of the distribution bundle.
+# If the file is missing please request a copy by contacting
+# jason@glencoesoftware.com.
+#
+
+from ... import SCHEMA_VERSION
+from .annotation import AnnotatableEncoder
+from omero.model import WellI
+
+
+class Well201501Encoder(AnnotatableEncoder):
+
+    TYPE = 'http://www.openmicroscopy.org/Schemas/OME/2015-01#Well'
+
+    def encode(self, obj):
+        v = super(Well201501Encoder, self).encode(obj)
+        self.set_if_not_none(v, 'Column', obj.column)
+        self.set_if_not_none(v, 'Row', obj.row)
+        self.set_if_not_none(v, 'ExternalDescription', obj.externalDescription)
+        self.set_if_not_none(v, 'ExternalIdentifier', obj.externalIdentifier)
+        self.set_if_not_none(v, 'Type', obj.type)
+        # FIXME: Color conversion required (uses integer encoded Color field)
+        self.set_if_not_none(v, 'Alpha', obj.alpha)
+        self.set_if_not_none(v, 'Red', obj.red)
+        self.set_if_not_none(v, 'Green', obj.green)
+        self.set_if_not_none(v, 'Blue', obj.blue)
+        self.set_if_not_none(v, 'omero:status', obj.status)
+
+        if obj.isWellSamplesLoaded() and obj.sizeOfWellSamples() > 0:
+            wellsamples = list()
+            for wellsample in obj.copyWellSamples():
+                wellsample_encoder = self.ctx.get_encoder(wellsample.__class__)
+                wellsamples.append(
+                    wellsample_encoder.encode(wellsample)
+                )
+            v['WellSamples'] = wellsamples
```

Comment:
However, I found that adding this breaks the unit tests because you get recursion! Encoding Plate -> encodes Wells -> Encodes Plate -> Encodes Wells etc!
This brings up a lot more questions like 'what direction of the graph does omero-marshal encode?'. E.g. if I encode a Tag with tagged Images loaded, do we also encode those images? If so, then we will get recursion. If not, then we miss out on that data/functionality.

The test recursion in the code above can be fixed with ```obj.plate.unloadWells()``` before encoding the plate, but what we really want is to know whether we're encoding the Well as part of a Plate (when we'd want to ignore the Well.plate entirely) OR if we're encoding the Well as a starting point, where we would want to encode the Plate (so we know the naming convention for the Well etc).